pkg/pillar/hardware/smart.go

```diff
@@ -80,6 +81,56 @@ func getInfoFromUnknownDisk(diskName, diskType string) *types.DiskSmartInfo {
 	return diskInfo
 }
 
+// getSmartType returns "Pre-fail" if the ATA attribute flag for pre-failure is set,
+// otherwise it returns "Old_age". According to the ATA spec, if bit 0 (prefailure) is set,
+// the attribute is considered a pre-fail attribute.
+// I have found this in smartmontools repo https://github.com/smartmontools/smartmontools.git
+// smartmontools/smartmontools/atacmds.h:164:#define ATTRIBUTE_FLAGS_PREFAILURE(x) (x & 0x01)
+// smartmontools/smartmontools/ataprint.cpp:1302: (ATTRIBUTE_FLAGS_PREFAILURE(attr.flags) ? "Pre-fail" : "Old_age"),
+func getSmartType(flags uint16) string {
```

Comment:
`flags` here is `uint16` but is `int` here: https://github.com/lf-edge/eve/pull/4671/files#diff-50e500f47382c3756300d83f5b548dab218fb516501fecad2cd3909458f3b66cR74

Reply:
@jsfakian it LGTM, but you still didn't answer this one :point_up:

Reply:
I changed that, didn't I?